cfme/infrastructure/config_management/__init__.py

```diff
@@ -496,226 +465,59 @@ def refresh_relationships(self, cancel=False):
     @property
     def config_profiles(self):
         """Returns 'ConfigProfile' configuration profiles (hostgroups) available on this manager"""
-        view = navigate_to(self, 'Details')
-        # TODO - remove it later.Workaround for BZ 1452425
-        view.toolbar.view_selector.select('List View')
-        view.toolbar.refresh.click()
-        wait_for(lambda: view.entities.elements.is_displayed, fail_func=view.toolbar.refresh.click,
-                 handle_exception=True, num_sec=60, delay=5)
-        config_profiles = []
-        for row in view.entities.elements:
-            if self.type == 'Ansible Tower':
-                name = row.name.text
-            else:
-                name = row.description.text
-            if 'unassigned' in name.lower():
-                continue
-            config_profiles.append(ConfigProfile(name=name, manager=self))
-        return config_profiles
+        return self.collections.config_profiles.all()
 
     @property
-    def systems(self):
+    def config_systems(self):
         """Returns 'ConfigSystem' configured systems (hosts) available on this manager"""
-        return sum([prof.systems for prof in self.config_profiles])
+        systems_per_prof = [prof.config_systems for prof in self.config_profiles]
+        return [item for sublist in systems_per_prof for item in sublist]
 
     @property
     def yaml_data(self):
         """Returns yaml data for this manager"""
         return conf.cfme_data.configuration_managers[self.key]
 
-    @classmethod
-    def load_from_yaml(cls, key):
-        """Returns 'ConfigManager' object loaded from yamls, based on its key"""
-        data = conf.cfme_data.configuration_managers[key]
-        creds = conf.credentials[data['credentials']]
-        return cls(
-            name=data['name'],
-            url=data['url'],
-            ssl=data['ssl'],
-            credentials=cls.Credential(
-                principal=creds['username'], secret=creds['password']),
-            key=key)
+    def from_yaml(self, key):
+        """ Given key, returns an object with yaml properties filled in"""
+        new_obj = copy(self)
```

Comment:
Q: why are copying this if simple instantiation possible with respect to class method?